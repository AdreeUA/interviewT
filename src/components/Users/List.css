.user-list__table {
  margin-bottom: 15px;
}
.user-list__text {
  margin-bottom: 15px;
}
