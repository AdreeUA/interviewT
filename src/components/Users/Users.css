.users {
  max-width: 1250px;
  margin: 0 auto;
  padding: 15px 0 30px;
}
.users__list {
  display: flex;
  margin: 0 -20px;
}
.users__item {
  flex: 1 1 50%;
  margin: 0 20px;
}
