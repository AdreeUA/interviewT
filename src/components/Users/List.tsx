import React from 'react'
import { Dispatch, connect } from 'react-redux'
import * as _ from 'lodash'
import { StoreState } from '../../store/types'
import { load as userLoad, UserState } from '../../ducks/users'

import './List.css'

interface OwnProps {}

interface ConnectedState {
  users: UserState[]
  isRequesting: boolean
}

interface ConnectedDispatch {
  dispatch: Dispatch<StoreState>
}

type Props = OwnProps & ConnectedState & ConnectedDispatch

interface State {}

const mapStateToProps = (state: StoreState) => ({
  users: state.users.results,
  isRequesting: state.users.isRequesting,
})

const WrappedList = connect<ConnectedState, ConnectedDispatch, OwnProps>(mapStateToProps)(
  class List extends React.Component<Props, State> {
    tableColumns = [
      {
        label: 'Image',
      },
      {
        label: 'Name',
      },
      {
        label: 'Email',
      },
      {
        label: 'Phone',
      },
    ]

    handleAddUser = () => {
      this.props.dispatch(userLoad())
    }

    render() {
      const { users } = this.props
      return (
        <div className="user-list">
          <table className="table user-list__table">
            <thead>
              <tr>
                {this.tableColumns.map((item) => {
                  return <th key={item.label}>{item.label}</th>
                })}
              </tr>
            </thead>
            <tbody>
              {users.map(({ fullName, thumbnail, email, phone }) => {
                return (
                  <tr key={email}>
                    <td>
                      <div className="table__image-wrapper">
                        <img src={thumbnail} className="table__image" alt={fullName} />
                      </div>
                    </td>
                    <td className="table__name">{fullName}</td>
                    <td>{email}</td>
                    <td>{phone}</td>
                  </tr>
                )
              })}
            </tbody>
          </table>
          {_.isEmpty(users) && !this.props.isRequesting && <div className="user-list__text text">Empty table</div>}
          {this.props.isRequesting && <div className="user-list__text text">Loading...</div>}
          <button className="button" onClick={this.handleAddUser}>
            Add user
          </button>
        </div>
      )
    }
  },
)

export default WrappedList
