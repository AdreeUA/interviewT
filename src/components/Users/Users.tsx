import React from 'react'
import List from './List'
import SortableTable from '../SortableTable'

import './Users.css'

interface Props {}

interface State {}

class Users extends React.Component<Props, State> {
  render() {
    return (
      <div className="users">
        <h2>Users</h2>
        <div className="users__list">
          <div className="users__item">
            <List />
          </div>
          <div className="users__item">
            <SortableTable />
          </div>
        </div>
      </div>
    )
  }
}

export default Users
