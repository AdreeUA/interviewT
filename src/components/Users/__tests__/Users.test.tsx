import * as React from 'react'
import { shallow } from 'enzyme'
import Users from '../Users'

describe('Users', () => {
  it('render() whith initialState', () => {
    const wrapper = shallow(<Users />)
    expect(wrapper).toMatchSnapshot()
  })
})
