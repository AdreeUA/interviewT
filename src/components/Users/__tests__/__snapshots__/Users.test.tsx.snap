// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Users render() whith initialState 1`] = `
<div
  className="users"
>
  <h2>
    Users
  </h2>
  <div
    className="users__list"
  >
    <div
      className="users__item"
    >
      <Connect(List) />
    </div>
    <div
      className="users__item"
    >
      <Connect(SortableTable) />
    </div>
  </div>
</div>
`;
